### PR TITLE
New chatbot examples api

### PR DESCRIPTION
**Pull request by aliabid94**
Spent a lot of time trying to simplify the chatbot examples API, and ended up with the below. 
Introduces zero new API for the user to learn, automatically embeds any gr.Examples that targets a chatbot inside the chatbot. No more `gr.Chatbot(suggestions=)`, `SuggestionMessage`, or `chatbot.suggestion_select` APIs for the user to learn. 
Still working on caching examples and triggering bot response so a draft.

```python
import gradio as gr

def add_textbox_message(history, message):
    history.append((message, None))
    return history, gr.Textbox(value=None, interactive=False)

def respond(history):
    print(history)
    history[-1][1] = "Cool, " + history[-1][0]
    return history

with gr.Blocks() as demo:
    chatbot = gr.Chatbot(scale=1)
    input_textbox = gr.Textbox(interactive=True, show_label=False)
    
    chat_msg_text = input_textbox.submit(add_textbox_message, [chatbot, input_textbox], [chatbot, input_textbox])
    bot_msg_text = chat_msg_text.then(respond, chatbot, chatbot)
    bot_msg_text.then(lambda: gr.Textbox(interactive=True), None, [input_textbox])

    examples = gr.Examples(
        [
            [[], "What's up"], 
            [[], "Try this example with this image."]
        ],
        inputs=[chatbot, input_textbox],
        outputs=[chatbot, input_textbox],
        fn=add_textbox_message
    )
```
Recording 2024-09-19 at 10 01 59

### CODE REVIEW

**Comment by dawoodkhan82** [comment]
It would be nice if the examples showed the image, in the multimodal case.

File: gradio/components/chatbot.py [hunk 1]
```diff
@@ -181,8 +168,9 @@ def __init__(
         likeable: bool = False,
         layout: Literal["panel", "bubble"] | None = None,
         placeholder: str | None = None,
-        suggestions: list[SuggestionMessage] | None = None,
         show_copy_all_button=False,
+        examples: list[str] | None=None,
+        show_examples_inside: bool = True,
```
Comment (dawoodkhan82) [inline gradio/components/chatbot.py:173]: Maybe rename this to `show_examples_in_chatbot`?

Comment (abidlabs) [reply gradio/components/chatbot.py:173]: Actually, why do we even need this parameter? No other component supports examples like this. If a user is passing in examples using the `examples` parameter of `Chatbot`, can't we assume they want examples inside the chatbot?

Comment (aliabid94) [reply gradio/components/chatbot.py:173]: hmm a bit confusing, but if you pass examples= directly, then they automatically render inside but don't do anything when clicked other than trigger a `load_example` listener. show_examples_inside=True means that any gr.Examples that outputs to this chatbot automatically is automatically rendered in the chatbot instead.

**Comment by aliabid94** [comment]
Changing a good bit of this, closing the PR for now

**Pull request closed without merging**